src/workerd/io/worker.c++

```diff
@@ -3565,4 +3570,13 @@ kj::Promise<WorkerInterface::CustomEvent::Result>
   return inner->customEvent(kj::mv(event));
 }
 
+int32_t Worker::getNextPromiseContextTag() const {
+  int32_t expected = kMaxSmi + 1;
```

Comment:
Would it make sense to use definitions from v8-internal.h? A Smi can be 31-bits (on 32-bit or with ptr compression) or 32-bits (on 64-bit). Just flagging this as an option. I've not gone digging but everything in v8 will know the rules for Smis. Tag could be defined in terms of `intptr_t` instead of `int32_t`. If the tag were incremented in the Smi domain then it can be done with just `fetch_add`.

I'm not sure the existing code works across multiple threads if this is the intent. Consider two threads calling this method and incrementing impl->promiseContextTag in close succession before either calls the compare-exchange.

Reply:
Updated!